internal/ingress/metric/collector/collector.go

```diff
@@ -273,6 +283,49 @@ func (sc *SocketCollector) Run() {
 	}
 }
 
+// RemoveMetrics deletes prometheus metrics from prometheus for endpoints that
+// are not available anymore.
+// Ref: https://godoc.org/github.com/prometheus/client_golang/prometheus#CounterVec.Delete
+func (sc *SocketCollector) RemoveMetrics(endpoints []string) {
+	mfs, err := prometheus.DefaultGatherer.Gather()
+	if err != nil {
+		glog.Errorf("error gathering metrics:", err)
+		return
+	}
+
+	toRemove := sets.NewString(endpoints...)
+
+	for _, mf := range mfs {
+		metricName := mf.GetName()
+		promHistogram, ok := sc.metricMapping[metricName]
+		if !ok {
+			continue
+		}
+
+		for _, m := range mf.GetMetric() {
+			labels := make(map[string]string, len(m.GetLabel()))
+			for _, labelPair := range m.GetLabel() {
+				labels[*labelPair.Name] = *labelPair.Value
+			}
+
+			endpoint, ok := labels["upstream_ip"]
+			if !ok {
+				continue
+			}
+
+			if !toRemove.Has(endpoint) {
+				continue
+			}
+
+			glog.Infof("Removing prometheus metric from histogram %v for endpoint %v", metricName, endpoint)
```

Comment:
Could be verbose on the default log level.

Reply:
done